### PR TITLE
Add markdownlint and fix violations

### PR DESCRIPTION
**Pull request by bmish**
Cleans up docs (especially regarding using consistent headers). Enforces markdownlint via linting/CI scripts.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
   "bin": "./bin/ember-template-lint.js",
   "scripts": {
     "changelog": "lerna-changelog",
-    "lint": "yarn lint:js",
+    "lint": "yarn lint:js && yarn lint:docs",
+    "lint:docs": "yarn markdownlint *.md dev/*.md docs/*.md docs/**/*.md -i CHANGELOG.md -i LICENSE.md",
```
Comment (rwjblue) [inline package.json]: For scripts that use bin scripts provided by packages, you do not have to invoke them with `yarn`. Specifically, in this case you would do:

```suggestion
    "lint:docs": "markdownlint *.md dev/*.md docs/*.md docs/**/*.md -i CHANGELOG.md -i LICENSE.md",
```

But also, is there any reason we can't use `markdownlint **/*.md` (e.g. why can't we run against all markdown files)?

**Comment by rwjblue** [comment]
(overall huge 👍 on this)

File: package.json (hunk 1 above)
Comment (bmish) [reply package.json]: Thanks, fixed both issues!

**Comment by rwjblue** [comment]
Thank you @bmish, this is awesome!

**Pull request merged**